lib/ansible/errors/__init__.py

```diff
@@ -85,15 +85,16 @@ def _get_error_lines_from_file(self, file_name, line_number):
 
         target_line = ''
         prev_line = ''
+        b_file_name = to_bytes(file_name, errors='surrogate_or_strict')
 
-        with open(file_name, 'r') as f:
+        with open(b_file_name, 'rb') as f:
             lines = f.readlines()
 
             target_line = lines[line_number]
             if line_number > 0:
                 prev_line = lines[line_number - 1]
 
-        return (target_line, prev_line)
+        return (to_native(target_line), to_native(prev_line))
```

Comment:
This should be either to_text or to_bytes.  In controller code we want our types to be well defined.  I currently see only one caller, and that caller is making the decision to use the surrogate_then_replace error handler because it knows the output will be used in a message to the user.  So I think returning bytes from this function is the correct action.

Reply:
This is being used for purposes of an exception.  I figured I'd stick with native.

Reply:
addressed